### ISSUE TITLE
differential_abundance.py failure

### ISSUE THREAD

**Issue by gregcaporaso** [biocore/qiime#1992]
I'm getting an error message that I can't interpret from ``differential_abundance.py``. @sowe9385, have you seen this before? I created my command by adapting the first usage example, and I confirmed that the BIOM table and the mapping file work when calling ``group_significance.py`` with the same category (so I know my BIOM table and mapping file are formatted correctly). This is using the latest development version of QIIME.

```bash
$ differential_abundance.py -i table_mc10534.no-positive-na.biom -o metagenomeSeq_fitZIG_diff_otus_Positive.txt -m ../../Cocci_soils_ITS_mapping_file.txt -a metagenomeSeq_fitZIG -c Positive -x Yes -y No
Traceback (most recent call last):
  File "/Users/karenschwarzberg/.virtualenvs/qiime19/bin/differential_abundance.py", line 159, in <module>
    main()
  File "/Users/karenschwarzberg/.virtualenvs/qiime19/bin/differential_abundance.py", line 142, in main
    DA_fitZIG(input_path, out_path, mapping_fp, mapping_category, subcategory_1, subcategory_2)
  File "/Users/karenschwarzberg/.virtualenvs/qiime19/lib/python2.7/site-packages/qiime/differential_abundance.py", line 61, in DA_fitZIG
    run_fitZIG(temp_fh.name, out_path, mapping_category, subcategory_1, subcategory_2)
  File "/Users/karenschwarzberg/.virtualenvs/qiime19/lib/python2.7/site-packages/qiime/differential_abundance.py", line 98, in run_fitZIG
    app_result = rsl(command_args=command_args, script_name='fitZIG.r')
  File "/Users/karenschwarzberg/.virtualenvs/qiime19/lib/python2.7/site-packages/qiime/util.py", line 1946, in __call__
    (exit_status, command, stdout, stderr))
burrito.util.ApplicationError: Unacceptable application exit status: 1
command: cd "/Users/karenschwarzberg/Documents/Analyses_for_others/Bridget/Apr-28_results/cdout_filtered/";  R --slave --args --source_dir /Users/karenschwarzberg/.virtualenvs/qiime19/lib/python2.7/site-packages/qiime/support_files/R -i /tmp/QIIME-differential-abundance-temp-table-IIq1Jr.biom -o metagenomeSeq_fitZIG_diff_otus_Positive.txt -c Positive -x Yes -y No  < /Users/karenschwarzberg/.virtualenvs/qiime19/lib/python2.7/site-packages/qiime/support_files/R/fitZIG.r
stdout:

stderr:
Error in if (!object$type %in% biomtypevals) { : 
  argument is of length zero
Calls: fitZIG ... initialize -> validObject -> anyStrings -> validityMethod
Execution halted


(qiime19)MacBook-Pro:cdout_filtered karenschwarzberg$ biom summarize-table -i table_mc10534.no-positive-na.biom
Num samples: 79
Num observations: 54457
Total count: 2742315
Table density (fraction of non-zero values): 0.039

Counts/sample summary:
 Min: 13823.0
 Max: 47577.0
 Median: 35230.000
 Mean: 34712.848
 Std. dev.: 6348.540
 Sample Metadata Categories: None provided
 Observation Metadata Categories: taxonomy

Counts/sample detail:
 T2.3d: 13823.0
 P3B: 18102.0
 T3B: 20467.0
 T2.6b: 21185.0
 AZ14.2Ab: 22756.0
 P6A: 24165.0
 T2.3a: 24696.0
 T2.6d: 27018.0
 T2.4a: 27337.0
 T2.6c: 27726.0
 AZ14.4Bb: 27821.0
 AZ14.5Bia: 27927.0
 T2.4c: 28890.0
 P2B: 29012.0
 AZ14.5Aia: 30355.0
 T2.10b: 30409.0
 AZ14.3Ab: 30448.0
 AZ14.5Biid: 30585.0
 AZ14.3Ba: 30673.0
 T2.3c: 31071.0
 T2.5c: 31124.0
 AZ14.5Biia: 31240.0
 P6B: 32240.0
 AZ14.5Aiib: 32873.0
 AZ14.4Ba: 33240.0
 AZ14.3Bc: 33382.0
 AZ14.5Aiia: 33523.0
 T2.2c: 33525.0
 T2.6a: 33819.0
 P2A: 33826.0
 T2.1a: 34034.0
 AZ14.2Bd: 34045.0
 AZ14.2Ba: 34211.0
 F3B: 34339.0
 T2.4b: 34520.0
 F3A: 34556.0
 AZ14.3Aa: 34612.0
 T5B: 34751.0
 P4B: 35084.0
 AZ14.5Aib: 35230.0
 T2B: 35272.0
 T2.4d: 35380.0
 T2.3b: 35835.0
 AZ14.5Bib: 36439.0
 T2.5d: 36543.0
 AZ14.3Bd: 36820.0
 P5B: 37114.0
 T2.2b: 37210.0
 T5A: 37383.0
 AZ14.5Biib: 37559.0
 AZ14.5Biic: 37612.0
 AZ14.5Bid: 37836.0
 AZ14.5Bic: 38056.0
 P3A: 38143.0
 T2.5a: 38167.0
 T2.10c: 38170.0
 AZ14.3Bf: 38170.0
 P5A: 38178.0
 AZ14.1f: 38242.0
 T2.10d: 38813.0
 T2.2d: 38893.0
 AZ14.2Bb: 39027.0
 P4A: 39633.0
 T2.5b: 39940.0
 T2.1b: 39964.0
 AZ14.3Be: 40086.0
 T2.2a: 40645.0
 T4A: 40713.0
 AZ14.1a: 40736.0
 T3A: 41305.0
 AZ14.2Bc: 41575.0
 T2A: 41847.0
 F2A: 41891.0
 T2.10a: 43035.0
 T4B: 43327.0
 T2.1d: 45091.0
 T2.1c: 45566.0
 F2B: 45882.0
 AZ14.2Aa: 47577.0
```

**Comment by gregcaporaso** [comment]
I also get a failure with ``DESeq2_nbinom``:

```bash
$ differential_abundance.py -i table_mc10534.no-positive-na.biom -o DESeq2_nbinom_diff_otus_Positive.txt -m ../../Cocci_soils_ITS_mapping_file.txt -a DESeq2_nbinom -c Positive -x Yes -y No
Traceback (most recent call last):
  File "/Users/karenschwarzberg/.virtualenvs/qiime19/bin/differential_abundance.py", line 159, in <module>
    main()
  File "/Users/karenschwarzberg/.virtualenvs/qiime19/bin/differential_abundance.py", line 150, in main
    DA_DESeq2(input_path, out_path, mapping_fp, mapping_category, subcategory_1, subcategory_2, DESeq2_diagnostic_plots)
  File "/Users/karenschwarzberg/.virtualenvs/qiime19/lib/python2.7/site-packages/qiime/differential_abundance.py", line 118, in DA_DESeq2
    run_DESeq2(temp_fh.name, out_path, mapping_category, subcategory_1, subcategory_2, DESeq2_diagnostic_plots, outfile_diagnostic) 
  File "/Users/karenschwarzberg/.virtualenvs/qiime19/lib/python2.7/site-packages/qiime/differential_abundance.py", line 159, in run_DESeq2
    app_result = rsl(command_args=command_args, script_name='DESeq2_nbinom.r')
  File "/Users/karenschwarzberg/.virtualenvs/qiime19/lib/python2.7/site-packages/qiime/util.py", line 1946, in __call__
    (exit_status, command, stdout, stderr))
burrito.util.ApplicationError: Unacceptable application exit status: 1
command: cd "/Users/karenschwarzberg/Documents/Analyses_for_others/Bridget/Apr-28_results/cdout_filtered/";  R --slave --args --source_dir /Users/karenschwarzberg/.virtualenvs/qiime19/lib/python2.7/site-packages/qiime/support_files/R -i /tmp/QIIME-differential-abundance-temp-table-9gEZtp.biom -o DESeq2_nbinom_diff_otus_Positive.txt -c Positive -x Yes -y No  < /Users/karenschwarzberg/.virtualenvs/qiime19/lib/python2.7/site-packages/qiime/support_files/R/DESeq2_nbinom.r
stdout:

stderr:
Error in if (!object$type %in% biomtypevals) { : 
  argument is of length zero
Calls: DESeq2_nbinom ... initialize -> validObject -> anyStrings -> validityMethod
Execution halted
```

**Comment by sowe9385** [comment]
Yes, I think I have seen it before.  Please try biom convert --to-json or
--to-hdf5 on your input biom table and try again?

**Comment by gregcaporaso** [comment]
Ok, we just converted it to json, and it seems to be working now. However, I had to specify the table type during the conversion, so I don't think the issue is the file type but that there is no table type in the original table (i.e., this is a duplicate of #1928 - I did search my error message, but I was searching the ``fitZIG`` variant, and that error message covers the ``CSS`` variant). Closing in favor of that error message - this is a critical fix for QIIME 1.9.1.
